### PR TITLE
Fix Script editor completion doesn't suggest members of a script for type hints

### LINKED ISSUE #47720: Script editor completion doesn't suggest members of a script after `:`
**Godot version:**

3.2.3

**OS/device including version:**

Linux

**Issue description:**

completion

completion1

**Steps to reproduce:**
1. Create a script `new_script.gd`:
```gdscript
class X:
	pass

```
2. Create another script with this code
```gdscript
const scr = preload("res://new_script.gd")
func _ready():
	var x: scr.
```
As you can see the completion doesn't appear
It should suggest `X`.

**Minimal reproduction project:**

### PR DESCRIPTION
**Pull request by ray90514**
Fix #47720 
result:
example

### CODE REVIEW

File: modules/gdscript/gdscript_editor.cpp [hunk 1]
```diff
@@ -2997,8 +2997,16 @@ Error GDScriptLanguage::complete_code(const String &p_code, const String &p_path
...
-					case GDScriptParser::DataType::SCRIPT:
 					case GDScriptParser::DataType::GDSCRIPT: {
+						Ref<GDScript> scr = base_type.script_type;
+						if (scr.is_valid()) {
+							for (const Map<StringName, Ref<GDScript> >::Element *E = scr->get_subclasses().front(); E; E = E->next()) {
+								ScriptCodeCompletionOption option(E->key().operator String(), ScriptCodeCompletionOption::KIND_CLASS);
+								options.insert(option.display, option);
+							}
+						}
+					}
```
Comment (vnen) [inline modules/gdscript/gdscript_editor.cpp:3008]: Doesn't this need an explicit `[[fallthrough]]`? Not sure why it's passing CI without it.

Comment (ray90514) [reply modules/gdscript/gdscript_editor.cpp:3009]: It's my fault. I added the macro.

**Comment by vnen** [comment]
Ah, just realized this is for 3.x. The fix looks good to me, but is this issue fixed in master already?

**Comment by ray90514** [comment]
> Ah, just realized this is for 3.x. The fix looks good to me, but is this issue fixed in master already?

The same script running in master will caused a parse-error, and the error message is 
`Parse Error: "scr" is a constant but does not contain a type.`

**Review by akien-mga — APPROVED** [review]
Sorry for the long delay!
Let's merge for 3.6 (and wait a bit for a potential 3.5.x cherry-pick to ensure there are no regressions).
I rebased the PR to make sure that it still compiles, since the last update was 18 months ago.

The bug this solves seems still valid in current 4.0 so this should be solved via a different PR.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.2.

**Pull request merged**